packages/puppeteer-core/src/common/FrameManager.ts

```diff
@@ -131,6 +131,9 @@ export class FrameManager extends EventEmitter {
     if (!mainFrame) {
       return;
     }
+    for (const child of mainFrame.childFrames()) {
```

Comment:
I think we need to set-up the listeners first as if we have a big tree we may miss it it happens too quickly.